### ISSUE TITLE
Feature to take backup of the existing secret before sync from external secret store

### ISSUE THREAD

**Issue by AnuragEkkati** [external-secrets/external-secrets#3146]
Option to enable ( can be disabled by default ) creating a backup of the synced secret.

It will be very useful if the operator can create a backup of the synced secret, which can be controlled by the externalsecrets CR. Can be false by default, but when configured "createBackup: true", the operator will create a backup of the secret before sync from the external secret store.

**Describe the solution you'd like**
externalsecrets CR should have a field to take backup of the existing secret before sync from the external secret store. 

"createBackup: true"

**Comment by Skarlso** [comment]
Hello!

Do you have a use case for this? Normally, these secrets should be throw-aways. Basically, it's syncing things into the cluster. Whatever the remote is, should be the source of truth.

**Comment by Skarlso** [comment]
Interesting. Certificates should be handled differently though but that's another discussion.

I mean, reverting is just changing the remote, isn't it? If your keys are backed up in remote, all you do is roll them back and ESO does its magic and applies the new secrets to the secret.

A secret created by eso is always ephemeral in nature since it can change continuously. The remote handles certificate rotation, roll back, right-ahead logs, auditing, etc. A Kubernetes secrets provides none of those features.

Anyways, I think ESO shouldn't be responsible for this, but if others think this feature would be a good thing, I, of course, will follow the majority. :)

**Comment by evrardj-roche** [comment]
I would disagree with this, it opens a can of worms about service levels (RTO/RPO) and security of the backups. 

The kubernetes administrator should have already policies in place for backuping etcd in a secure fashion.

**Comment by AnuragEkkati** [comment]
We can close this issue. Thanks for the inputs.

**Comment by evrardj-roche** [comment]
Thanks @AnuragEkkati !
